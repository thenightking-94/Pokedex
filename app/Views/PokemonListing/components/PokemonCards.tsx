import React, { ReactElement, useState} from 'react';
import { useQuery } from 'react-query';
import { getPokemonDetails, QueryKeys } from '../../../React-query-middleware';
import { Bars } from 'react-loader-spinner'
import { getSuitableColorCodes, formatPokemonId, formatPokemonText, returnLinearGradientStyles } from '../../../Utils/Utils';
import { useRouter } from 'next/router';
import { useMediaQuery } from '@mui/material';

export default function PokemonCards (props): ReactElement{
    const {item : { name, url}, inDetailPage} = props;
    const isMobile = useMediaQuery('(max-width:720px)')
    const PokemonId : string = (url.trim().split('/'))[6] as string
    const router = useRouter()
    const {data, isLoading, error} = useQuery([QueryKeys.getPokemonDetails, PokemonId], () => getPokemonDetails(PokemonId))
    let imgUrl :string = data?.sprites?.other?.dream_world?.front_default as string
    let pokemontype : Array<string> = data?.types?.map(i => i.type.name) as Array<string>
    const [color1, color2, ...rest] =  getSuitableColorCodes(pokemontype)
    
   const goToDetailPage = () =>{
    router.push(`/detail/${PokemonId}`)
   }

    return(
        <>
            <div onClick={goToDetailPage} className={(isMobile && inDetailPage) ? 'minified-card' : 'pokemon-card'}>
                {isLoading && <Bars height="60" width="60" color="grey" ariaLabel="bars-loading" visible={true} /> }
                {
                    !!data && 
                    <div className='pokemon-data' style={returnLinearGradientStyles(color1, color2)}>
                         <img alt={name} aria-labelledby={name} src={imgUrl} className='list-img'/>
                         <p className='label'>{formatPokemonText(name)}</p>
                         <p className='id-label'>{formatPokemonId(PokemonId)}</p>
                    </div>
                }
            </div>
        </>    
    )

}