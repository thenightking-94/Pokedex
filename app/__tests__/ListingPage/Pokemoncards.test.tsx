import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import renderer from "react-test-renderer";
import PokemonCards from '../../Views/PokemonListing/components/PokemonCards'

const queryClient = new QueryClient()
const props = {
   item:  {name : 'Charizard', url: 'https://pokeapi.co/api/v2/pokemon/6/'}
}

describe('PokemonListing', () => {
  it('renders correctly', () => {
    render(
    <QueryClientProvider client={queryClient}>
         <PokemonCards {...props}/>
    </QueryClientProvider>
   )
  })
})

describe("Jest Snapshot testing suite", () => {
    it("Matches DOM Snapshot", () => {
      const domTree = renderer.create( 
      <QueryClientProvider client={queryClient}>
        <PokemonCards {...props}/>
     </QueryClientProvider>).toJSON();
      expect(domTree).toMatchSnapshot();
    });
  });