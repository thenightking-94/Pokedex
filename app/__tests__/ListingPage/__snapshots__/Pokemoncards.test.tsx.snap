// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  className="pokemon-card"
  onClick={[Function]}
>
  <div
    aria-busy={true}
    aria-label="bars-loading"
    data-testid="bars-loading"
    role="status"
    style={
      {
        "display": "flex",
      }
    }
  >
    <svg
      data-testid="bars-svg"
      fill="grey"
      height="60"
      viewBox="0 0 135 140"
      width="60"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="120"
        rx="6"
        width="15"
        y="10"
      >
        <animate
          attributeName="height"
          begin="0.5s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="120;110;100;90;80;70;60;50;40;140;120"
        />
        <animate
          attributeName="y"
          begin="0.5s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="10;15;20;25;30;35;40;45;50;0;10"
        />
      </rect>
      <rect
        height="120"
        rx="6"
        width="15"
        x="30"
        y="10"
      >
        <animate
          attributeName="height"
          begin="0.25s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="120;110;100;90;80;70;60;50;40;140;120"
        />
        <animate
          attributeName="y"
          begin="0.25s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="10;15;20;25;30;35;40;45;50;0;10"
        />
      </rect>
      <rect
        height="140"
        rx="6"
        width="15"
        x="60"
      >
        <animate
          attributeName="height"
          begin="0s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="120;110;100;90;80;70;60;50;40;140;120"
        />
        <animate
          attributeName="y"
          begin="0s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="10;15;20;25;30;35;40;45;50;0;10"
        />
      </rect>
      <rect
        height="120"
        rx="6"
        width="15"
        x="90"
        y="10"
      >
        <animate
          attributeName="height"
          begin="0.25s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="120;110;100;90;80;70;60;50;40;140;120"
        />
        <animate
          attributeName="y"
          begin="0.25s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="10;15;20;25;30;35;40;45;50;0;10"
        />
      </rect>
      <rect
        height="120"
        rx="6"
        width="15"
        x="120"
        y="10"
      >
        <animate
          attributeName="height"
          begin="0.5s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="120;110;100;90;80;70;60;50;40;140;120"
        />
        <animate
          attributeName="y"
          begin="0.5s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="10;15;20;25;30;35;40;45;50;0;10"
        />
      </rect>
    </svg>
  </div>
</div>
`;
